app/models/usage.rb

```diff
@@ -52,15 +52,41 @@ def emission
     amount_used * conversion_factor
   end
 
+  def daily_emission
+    emission / number_days
+  end
+
   def self.user_total_emissions(user)
     usages = Usage.user_usage(user)
     total_emissions = 0
-    usages.each { |usage| total_emissions += usage.emission }
+    usages.each {|usage| total_emissions += usage.emission}
     total_emissions
   end
 
-  def daily_emission
-    emission / number_days
+  def self.usages_by_date(user, date)
+    Usage.where("user_id = ? AND start_date <= ? AND end_date >= ?",
+      user,
+      date,
+      date)
+  end
+
+  def self.total_daily_emissions(user, date)
+    usages = Usage.usages_by_date(user, date)
+    daily_total = 0
+    usages.each  {|usage| daily_total += usage.daily_emission}
```

Comment:
Unnecessary spacing detected.<br>Space between { and | missing.<br>Space missing inside }.